### PR TITLE
Db/ar 19/mettre en place les schemas de db

### PR DESCRIPTION
**Pull request by badidu09**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [ ] I have documented my code
- [x] Do we need to implement unit tests?

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **Describe your changes**
Add mysql db & create all schema
* **Issue ticket number and link**

## Summary by CodeRabbit

- **New Features**
  - Implemented user account creation with a new user registration endpoint.
  - Introduced new models for actions, reactions, and associated data, enhancing the application's functionality.

- **Enhancements**
  - Expanded the database configuration to support environment variables for improved security and configurability.
  - Refined model properties to allow null values where appropriate, providing greater flexibility in data handling.

- **Documentation**
  - Updated API documentation to include the new user creation route.

- **Bug Fixes**
  - Addressed issues with model synchronization to ensure accurate database schema representation.

Please note that these changes may require you to update your application settings or interact with new features.

### CODE REVIEW

File: back/src/controller/user.controller.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      username,
+      password,
+    });
+    res.status(201).json(user);
+  } catch (error: any) {
+    res.status(500).json({ error: error.message });
+  }
+}
+
+exports.createUser = createUser;
```
Comment (romainpanno) [inline back/src/controller/user.controller.ts:17]: EOF

Comment (romainpanno) [reply back/src/controller/user.controller.ts:17]:
```suggestion
exports.createUser = createUser;

```

File: back/src/index.ts [hunk 2]
```diff
@@ -13,6 +15,8 @@ app.get('/', (req, res) => {
   res.send('Hello World!');
 });
 
+app.use('/users', UserRouter);
```
Comment (romainpanno) [inline back/src/index.ts:18]: You mentioned "comments" last time for routes, is it interesting to add it now ?

File: back/src/models/action.model.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      },
+      {
+        sequelize,
+        tableName: 'actions',
+      }
+    );
+  }
+}
+
+export { Action };
```
Comment (romainpanno) [inline back/src/models/action.model.ts:42]: EOF

Comment (romainpanno) [reply back/src/models/action.model.ts:42]:
```suggestion
export { Action };

```

File: back/src/models/actionData.model.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      },
+      {
+        sequelize,
+        tableName: 'actionsData',
+      }
+    );
+  }
+}
+
+export { ActionData };
```
Comment (romainpanno) [inline back/src/models/actionData.model.ts:37]: EOF
```suggestion
export { ActionData };

```

File: back/src/models/reaction.model.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      },
+      {
+        sequelize,
+        tableName: 'reactions',
+      }
+    );
+  }
+}
+
+export { Reaction };
```
Comment (romainpanno) [inline back/src/models/reaction.model.ts:42]: EOF
```suggestion
export { Reaction };

```

File: back/src/models/reaction.model.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+// action.model.ts
```
Comment (romainpanno) [inline back/src/models/reaction.model.ts:1]: ??

File: back/src/models/actionData.model.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
+// action.model.ts
```
Comment (romainpanno) [inline back/src/models/actionData.model.ts:1]: ??

File: back/src/models/reactionData.model.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
+// action.model.ts
```
Comment (romainpanno) [inline back/src/models/reactionData.model.ts:1]: ??

File: back/src/models/reactionData.model.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      },
+      {
+        sequelize,
+        tableName: 'reactionData',
+      }
+    );
+  }
+}
+
+export { ReactionData };
```
Comment (romainpanno) [inline back/src/models/reactionData.model.ts:37]: EOF
```suggestion
export { ReactionData };

```

File: back/src/models/service.model.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+      {
+        sequelize,
+        tableName: 'services',
+      }
+    );
+  }
+  
+}
+
+export { Service };
```
Comment (romainpanno) [inline back/src/models/service.model.ts:43]: EOF
```suggestion
export { Service };

```

File: back/src/route/user.route.ts [hunk 11]
```diff
@@ -0,0 +1,8 @@
+import express from 'express';
+const router = express.Router();
+
+const userController = require('../controller/user.controller');
+
+router.post('/create', userController.createUser);
+
+module.exports = router;
```
Comment (romainpanno) [inline back/src/route/user.route.ts:8]: EOF
```suggestion
module.exports = router;

```

**Review by romainpanno — CHANGES_REQUESTED** [review]

File: back/src/index.ts (hunk 2 above)
Comment (badidu09) [reply back/src/index.ts:18]: for the moment no because this route ll change in next week

**Review by romainpanno — APPROVED** [review]
OK

**Review by Beafowl-Pull — APPROVED** [review]

**Review by Vivelis — APPROVED** [review]

**Review by romainpanno — DISMISSED** [review]

File: back/src/models/service.model.ts [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+        },
+      },
+      {
+        sequelize,
+        tableName: 'services',
+      }
+    );
+  }
+  
+}
```
Comment (Vivelis) [inline back/src/models/service.model.ts:41]: should be no back to line 
```suggestion
  }
}
```

File: back/src/models/user.model.ts [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+      {
+        sequelize,
+        tableName: 'users',
+      }
+    );
+  }
+}
+  
+
+export { User };
```
Comment (Vivelis) [inline back/src/models/user.model.ts:37]: Should be only one back to line
```suggestion
}

export { User };
```

**Review by Vivelis — CHANGES_REQUESTED** [review]
Problems of style

**Review by Beafowl-Pull — APPROVED** [review]

**Review by romainpanno — APPROVED** [review]

**Pull request merged**